src/commands/dsyms/utils.ts

```diff
@@ -0,0 +1,127 @@
+import {exec} from 'child_process'
+import {mkdirSync} from 'fs'
+import glob from 'glob'
+import path from 'path'
+import {promisify} from 'util'
+import {buildPath} from '../../helpers/utils'
+import {Payload} from './interfaces'
+import {checkNonEmptyFile} from './validation'
+
+const UUID_REGEX = '[0-9A-F]{8}-([0-9A-F]{4}-){3}[0-9A-F]{12}'
+
+const getMatchingDSYMFiles = async (absoluteFolderPath: string): Promise<Payload[]> => {
+  const dSYMFiles = glob.sync(buildPath(absoluteFolderPath, '**/*.dSYM'))
+
+  return Promise.all(
+    dSYMFiles.map(async (dSYMPath) => {
+      const uuids = await dwarfdumpUUID(dSYMPath)
+
+      return {
+        path: dSYMPath,
+        type: 'ios_symbols',
+        uuids,
+      }
+    })
+  )
+}
+
+export const getPayloads = async (searchPaths: string[]): Promise<Payload[]> => {
+  const nestedPayloads = await Promise.all(searchPaths.map((searchPath: string) => getMatchingDSYMFiles(searchPath)))
+
+  return nestedPayloads.flat()
+}
+
+export const getSearchPaths = async (baseSearchPath: string): Promise<string[]> => {
+  const zipFiles = getZipFiles(baseSearchPath)
+  const unzipFolder = buildPath('/tmp', 'datadog-ci', 'dsyms')
+  const searchPaths = (await Promise.all(
+    zipFiles
+      .map(async (zipFile: string) => {
+        const unzipPath = buildPath(unzipFolder, path.basename(zipFile, '.zip'), Date.now().toString())
+        try {
+          await unzip(zipFile, unzipPath)
+
+          return unzipPath
+        } catch (err) {
+          console.log(err)
```

Comment:
Is this how we'll handle the errors?

Reply:
i went with the simplest way that i knew. 

open to suggestions: should i pass `this.context.stderr` from `upload.ts` and use it here?

Reply:
I think it would be more user friendly to let the error throw here and deal with it in `upload.ts` and craft an error message that the user will understand.

Reply:
we may have one single error among multiple zip files, the error case shouldn't stop this function.
then i need to return both `searchPaths` and `errors` from this function, is this okay?

```
...
return { searchPaths: searchPaths, errors: errors }
```

Reply:
sure.

Reply:
https://github.com/DataDog/datadog-ci/pull/300#discussion_r680956626
now we are reverting back to "either folder or a zip path": we won't be unzipping every zip file.
making this change will simplify `getSearchPaths` function and we won't need to return both paths and errors.